ee_plugin/ee_plugin.py

```diff
@@ -209,10 +209,17 @@ def unload(self):
         if self.menu:
             self.iface.pluginMenu().removeAction(self.menu.menuAction())
 
-        if self.toolButton:
-            self.toolButton.deleteLater()
+        try:
+            if self.toolButton:
+                self.toolButton.deleteLater()
+        except RuntimeError as e:
+            print(f"Error deleting toolButton: {e}")
 
-        QgsApplication.processingRegistry().removeProvider(self.provider)
+        if (
+            self.provider
+            and self.provider in QgsApplication.processingRegistry().providers()
+        ):
```

Comment:
```suggestion
        if self.provider in QgsApplication.processingRegistry().providers():
```

Slight opportunity to simplify.  I wouldn't imagine `None in QgsApplication.processingRegistry().providers()` would ever be truthy.